### PR TITLE
fix(file-uploader): Added better error text for over max files

### PR DESCRIPTION
**Pull request by ErikCH**
#### Description of changes
The maximum number of files error message was not clear. Added more explicit text.

Updated error text
image

#### Description of how you validated changes

Updated text for test that tests out message when user is over max file limit

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react/src/components/Storage/FileUploader/UploadPreviewer/index.tsx
Comment (ErikCH) [inline packages/react/src/components/Storage/FileUploader/UploadPreviewer/index.tsx:27]: When I redo the translations this will be updated.

File: packages/react/src/components/Storage/FileUploader/__tests__/FileUploader.test.tsx
Comment (ErikCH) [inline packages/react/src/components/Storage/FileUploader/__tests__/FileUploader.test.tsx:605]: This test already tests the text, I just updated it.

File: packages/react/src/components/Storage/FileUploader/UploadPreviewer/index.tsx
Comment (ErikCH) [reply packages/react/src/components/Storage/FileUploader/UploadPreviewer/index.tsx]: +1 [11394eb](https://github.com/aws-amplify/amplify-ui/pull/3344/commits/11394ebebbc792ddb8299e074c7569ad2e35a103)
Merge state
Add more commits by pushing to the [fileuploader/over-max-file-error](https://github.com/aws-amplify/amplify-ui/tree/fileuploader/over-max-file-error) branch on [aws-amplify/amplify-ui](https://github.com/aws-amplify/amplify-ui).

This branch was previously deployed
1 inactive (outdated) deployment
Code owner review required
Waiting on code owner review from aws-amplify/amplify-ui. 
1 pending reviewer
Some checks haven’t completed yet
2 in progress, 2 successful, and 11 expected checks
@github-actions
CodeQL / Analyze (javascript) (pull_request) In progress — This check has started...
[Details](https://github.com/aws-amplify/amplify-ui/actions/runs/4001669743/jobs/6868141330)
@github-actions
Test / PRs / setup (pull_request_target) Successful in 2s
[Details](https://github.com/aws-amplify/amplify-ui/actions/runs/4001669618/jobs/6868141049)
@github-actions
Test / PRs / dependency-review (pull_request_target) Successful in 7s
[Details](https://github.com/aws-amplify/amplify-ui/actions/runs/4001669618/jobs/6868141225)
@github-actions
Test / PRs / test / setup (pull_request_target) In progress — This check has started...
[Details](https://github.com/aws-amplify/amplify-ui/actions/runs/4001669618/jobs/6868142991)
CodeQL Expected — Waiting for status to be reported
Required
test / docs Expected — Waiting for status to be reported
Required
test / e2e (angular, angular, @angular and not @todo-angular) Expected — Waiting for status to be reported
Required
test / e2e (next, react, @react and not @todo-react) Expected — Waiting for status to be reported
Required
test / e2e (vue, vue, @vue and not @todo-vue) Expected — Waiting for status to be reported
Required
test / unit (angular) Expected — Waiting for status to be reported
Required
test / unit (react) Expected — Waiting for status to be reported
Required
test / unit (react-core) Expected — Waiting for status to be reported
Required
test / unit (react-native) Expected — Waiting for status to be reported
Required
test / unit (ui) Expected — Waiting for status to be reported
Required
test / unit (vue) Expected — Waiting for status to be reported
Required
This branch is out-of-date with the base branch
Merge the latest changes from main into this branch.
This merge commit will be associated with icystorm@gmail.com.

Merging is blocked
Merging can be performed automatically with 2 approving reviews.
 Automatically merge when all requirements are met. [Learn more](https://docs.github.com/github/collaborating-with-issues-and-pull-requests/automatically-merging-a-pull-request)
@ErikCH

 
Add heading textAdd bold text, <Cmd+b>Add italic text, <Cmd+i>
Add a quote, <Cmd+Shift+.>Add code, <Cmd+e>Add a link, <Cmd+k>
Add a bulleted list, <Cmd+Shift+8>Add a numbered list, <Cmd+Shift+7>Add a task list, <Cmd+Shift+l>
Directly mention a user or team
Reference an issue, pull request, or discussion
Add saved reply
Leave a comment
No file chosen
Attach files by dragging & dropping, selecting or pasting them.
Styling with Markdown is supported
Remember, contributions to this repository should follow its [contributing guidelines](https://github.com/aws-amplify/amplify-ui/blob/68c0a0e1c89ede36b4b0bdd7aad8f82f8c5b23b7/CONTRIBUTING.md), [security policy](https://github.com/aws-amplify/.github/blob/677e2710f7c81179e1b20e2f014a1b96ef0cbc35/SECURITY.md), and [code of conduct](https://github.com/aws-amplify/amplify-ui/blob/68c0a0e1c89ede36b4b0bdd7aad8f82f8c5b23b7/.github/CODE_OF_CONDUCT.md).
 ProTip! Add comments to specific lines under [Files changed](https://github.com/aws-amplify/amplify-ui/pull/3344/files).
Reviewers
@[wlee221](https://github.com/wlee221)
wlee221
@aws-amplify/amplify-ui
amplify-ui
At least 2 approving reviews are required to merge this pull request.

Still in progress? 
Assignees
No one—
Labels
None yet
Projects
None yet
Milestone
No milestone
Development
Successfully merging this pull request may close these issues.

None yet

Notifications
Customize
You’re receiving notifications because you authored the thread.
2 participants
@ErikCH
@wlee221

File: .changeset/purple-hotels-tease.md
Comment (reesscot) [inline .changeset/purple-hotels-tease.md:5]:
```suggestion
fix: Updated error text for max file count to be more explicit.
```

File: packages/react/src/components/Storage/FileUploader/FileUploader.tsx
Comment (reesscot) [inline packages/react/src/components/Storage/FileUploader/FileUploader.tsx:382]: `maxFiles` seems too vague. How about `maxFileCount`?

File: packages/react/src/components/Storage/FileUploader/UploadPreviewer/index.tsx
Comment (reesscot) [inline packages/react/src/components/Storage/FileUploader/UploadPreviewer/index.tsx:25]: I think this error message could be improved. How about something like:
`Cannot choose more than ${maxFiles}`

Comment (ioanabrooks) [reply packages/react/src/components/Storage/FileUploader/UploadPreviewer/index.tsx:25]: +1, as is translating it to other languages will lead to awkward phrasings.

File: packages/react/src/components/Storage/FileUploader/FileUploader.tsx
Comment (ioanabrooks) [reply packages/react/src/components/Storage/FileUploader/FileUploader.tsx:382]: renaming `maxFiles` is a breaking change now I believe.

File: packages/react/src/components/Storage/FileUploader/UploadPreviewer/index.tsx
Comment (reesscot) [reply packages/react/src/components/Storage/FileUploader/UploadPreviewer/index.tsx:25]: Ooh, good call @ioanabrooks on the translations aspect

File: packages/react/src/components/Storage/FileUploader/FileUploader.tsx
Comment (reesscot) [reply packages/react/src/components/Storage/FileUploader/FileUploader.tsx:382]: This component is in dev preview, so breaking changes are allowed

Comment (ErikCH) [reply packages/react/src/components/Storage/FileUploader/FileUploader.tsx:382]: Could we separate the renaming of `maxFiles` to `maxFilesCount` into a different PR? The renaming of a prop will affect several files, and it's a bit out of scope for this PR. I'll add a ticket to do it though before GA.

File: .changeset/purple-hotels-tease.md
Comment (ErikCH) [reply .changeset/purple-hotels-tease.md:5]: 👍

File: packages/react/src/components/Storage/FileUploader/UploadPreviewer/index.tsx
Comment (ErikCH) [reply packages/react/src/components/Storage/FileUploader/UploadPreviewer/index.tsx:25]: +1

Comment (ErikCH) [reply packages/react/src/components/Storage/FileUploader/UploadPreviewer/index.tsx]: [613ecbb](https://github.com/aws-amplify/amplify-ui/pull/3344/commits/613ecbbec83882cbffd7e56b105c839097dc47d1)

File: .changeset/purple-hotels-tease.md
Comment (ErikCH) [reply .changeset/purple-hotels-tease.md]: [613ecbb](https://github.com/aws-amplify/amplify-ui/pull/3344/commits/613ecbbec83882cbffd7e56b105c839097dc47d1)

File: packages/react/src/components/Storage/FileUploader/types.ts
Comment (ioanabrooks) [inline packages/react/src/components/Storage/FileUploader/types.ts:42]: `maxFiles` is not required on `FileUploaderProps`, have we tested edge cases? What happens if it's not specified by customers?

**Review by ioanabrooks — APPROVED** [review]

File: packages/react/src/components/Storage/FileUploader/types.ts
Comment (ioanabrooks) [reply packages/react/src/components/Storage/FileUploader/types.ts:42]: Thanks for confirming!

File: packages/react/src/components/Storage/FileUploader/UploadPreviewer/index.tsx
Comment (wlee221) [reply packages/react/src/components/Storage/FileUploader/UploadPreviewer/index.tsx]: I think the wording could be further improved once we have better support for translations with dynamic string, with something like

```
Cannot choose more than ${maxFiles} files
```

I think that looks better in English at least. Not sure if we want to do this now or later though, since our current support dynamic phrases are bad. What do you think? @ioanabrooks @ErikCH @reesscot

File: packages/react/src/components/Storage/FileUploader/types.ts
Comment (reesscot) [reply packages/react/src/components/Storage/FileUploader/types.ts:42]: Is there any S3 limit to how many files can be uploaded at a time? I assume we handle that gracefully?

Comment (ErikCH) [reply packages/react/src/components/Storage/FileUploader/types.ts:42]: It seems s3 should scale with load. So you can do up to 3500 PUTs per second according to this [article](https://docs.aws.amazon.com/AmazonS3/latest/userguide/optimizing-performance.html). But you could perhaps scale that more. 

I want to say here the bottle neck would be the browser, and the OS for the user.

File: packages/react/src/components/Storage/FileUploader/UploadPreviewer/index.tsx
Comment (ErikCH) [reply packages/react/src/components/Storage/FileUploader/UploadPreviewer/index.tsx]: This feels like something that can be looked at again, after we get translations in. I think we don't want to add the `files` to the end because of the translations.

Comment (wlee221) [reply packages/react/src/components/Storage/FileUploader/UploadPreviewer/index.tsx]: Understood, let's discuss in future translations pr then.

**Review by wlee221 — APPROVED** [review]

**Pull request merged**
